src/NLog/Targets/EventLogTarget.cs

```diff
@@ -260,8 +265,17 @@ protected override void Write(LogEventInfo logEvent)
                 category = Convert.ToInt16(this.Category.Render(logEvent), CultureInfo.InvariantCulture);
             }
 
+            long maxKilobytes = 0;
+
+            if(this.MaxKilobytes != null)
+            {
+                maxKilobytes = Convert.ToInt64(this.MaxKilobytes.Render(logEvent), CultureInfo.InvariantCulture);
+            }
+
             EventLog eventLog = GetEventLog(logEvent);
 
+            eventLog.MaximumKilobytes = maxKilobytes;
```

Comment:
if `this.MaxKilobytes` is not set, then this will set it to 0. ;)

Therefor the failing unit tests